src/librustc_metadata/encoder.rs

```diff
@@ -1435,6 +1458,8 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             def_path_table: def_path_table,
             impls: impls,
             exported_symbols: exported_symbols,
+            compiler_working_dir: tcx.sess.working_dir.to_string_lossy().into_owned(),
+            debug_prefix_map: debug_prefix_map,
```

Comment:
I think this problem could be solved by adapting the rules slightly. We could say that 

* if you specify *any* debug-prefix-map then debuginfo for inlined functions just does not work anymore out-of-the-box (you can still fix that manually of course).
* Conversely, if you do not specify a debug-prefix-map then your builds will not be reproducible. 

If we do it like this we should be able to transform the codemap before storing it in crate metadata and thus would not need to store the working dir and prefix mapping.

I think this would work well in practice.